### PR TITLE
Conditional SAML dependencies (fix Heroku deploys)

### LINKED ISSUE #5862: PostHog doesn't build on Heroku because of `xmlsec`
## Bug description

Since https://github.com/PostHog/posthog/pull/5681 added Python package [`xmlsec`](https://pypi.org/project/xmlsec/) as a dependency, PostHog doesn't build on Heroku, because that package is bindings for a C library that needs to be installed on the system.

See this build log from a review app:
Screen Shot 2021-09-08 at 19 46 50

@paolodamico 

## How to reproduce

1. Try to launch PostHog on Heroku from `master`.

### PR DESCRIPTION
**Pull request by paolodamico**
## Changes

This PR changes how we install SAML dependencies to only run conditionally if `SAML_AVAILABLE` variable is set. This way you don't need to install all the `xmlsec` C dependencies which can be problematic. This will fix Heroku deployments too. Closes #5862.

## Checklist

- [ ] All querysets/queries filter by Organization, by Team, and by User
- [ ] Django backend tests
- [ ] Jest frontend tests
- [ ] Cypress end-to-end tests
- [ ] Migrations are safe to run at scale (e.g. PostHog Cloud) – present proof if not obvious
- [ ] New/changed UI is decent on smartphones (viewport width around 360px)

### CODE REVIEW

File: .github/workflows/docker-image-publish.yml [hunk 1]
```diff
@@ -36,6 +36,8 @@ jobs:
               id: docker-latest
               if: github.repository == 'PostHog/posthog'
               uses: docker/build-push-action@v2
+              env:
+                  SAML_AVAILABLE: '1'
```
Comment (paolodamico) [inline .github/workflows/docker-image-publish.yml:40]: You can see that default Docker images will always include SAML packages

File: app.json [hunk 2]
```diff
@@ -16,7 +16,7 @@
                 "DEPLOYMENT": "Heroku Review App",
                 "AUTO_LOGIN": true
             },
-            "buildpacks": [{ "url": "heroku-community/apt" }, { "url": "heroku/nodejs" }, { "url": "heroku/python" }],
+            "buildpacks": [{ "url": "heroku/nodejs" }, { "url": "heroku/python" }],
```
Comment (paolodamico) [inline app.json:19]: We no longer attempt to install `xmlsec` deps on Heroku. SAML won't run on Heroku (documented).

File: .github/workflows/ci-backend.yml [hunk 3]
```diff
@@ -118,6 +115,61 @@ jobs:
...
                   CLICKHOUSE_VERIFY: 'False'
+              run: |
+                  mkdir -p frontend/dist
+                  touch frontend/dist/index.html
+                  touch frontend/dist/layout.html
+                  touch frontend/dist/shared_dashboard.html
+                  pytest ee -m "not saml_only"
+
+    saml:
+        name: Django tests - with SAML
```
Comment (Twixes) [inline .github/workflows/ci-backend.yml:126]: Nit: this should have an en dash (`–`) for consistency

**Review by Twixes — COMMENTED** [review]
This seems to make sense!

File: .github/workflows/ci-backend.yml [hunk 4]
```diff
@@ -118,6 +115,61 @@ jobs:
...
+
+            - name: Run posthog tests
+              env:
+                  DATABASE_URL: 'postgres://posthog:posthog@localhost:5432/posthog'
+              run: |
+                  mkdir -p frontend/dist
+                  touch frontend/dist/index.html
+                  touch frontend/dist/layout.html
+                  touch frontend/dist/shared_dashboard.html
+                  pytest posthog/
```
Comment (macobo) [inline .github/workflows/ci-backend.yml:164]: Rather than running everything, worth running only things with some specific pytest.mark?

Comment (paolodamico) [reply .github/workflows/ci-backend.yml:164]: Not sure if it's still worth running all tests as this is the only context where we run them with all the extra SAML dependencies, making sure that having them doesn't break something down.

**Pull request merged**